### ISSUE TITLE
[Shining Force 3] Various graphical issues

### ISSUE THREAD

**Issue by barbudreadmon** [FCare/Kronos#378]
I'm creating a dedicated issue about SF3, also mentioning #181 #339 #341 

Masqurin's blaze spell issue is bad, the other ones are minor. 

Quoting an awesome report on the libretro forum : 
>I just tried it, I’m on chapter 4, some spells render fine, the overall graphics are OK, Grace’s healing spell renders almost perfectly, apart of the aura effect, amazingly, the aura effects in Scenario 2 and 3 are rendered correctly, since they are different from SC1, also note the nice patch I applied from Meduza team, it will remove the black borders from the 3 scenarios and other games too.

![img1](https://forums.libretro.com/uploads/default/optimized/2X/8/8db19ceeec623ffa329f46dc0ac9b7d37fc74846_1_690x388.jpg)

>Masqurin’s spells render correctly, I mean thunder and wind ones, the fire spell is really broken here, note right before the spell it’s OK, then broken, once the magic is finished, the rendering back to normal.

![img2](https://forums.libretro.com/uploads/default/optimized/2X/e/e7a870987ae878fd7304b1bf0d0e37c1b170897f_1_690x388.jpg)

>Broken

![img3](https://forums.libretro.com/uploads/default/optimized/2X/d/dff459b82266dbfad5d0a588ad7e146cf3146e40_1_690x388.jpg)

![img4](https://forums.libretro.com/uploads/default/optimized/2X/b/b3e1b7a1a1049769929067edc777323b55399dbc_1_690x388.jpg)

>Only the aura here is weird

![img5](https://forums.libretro.com/uploads/default/optimized/2X/0/0c025e25e36dccc5d75491747303a9edad857e01_1_690x388.jpg)

>This scene is properly rendered

![img6](https://forums.libretro.com/uploads/default/optimized/2X/b/b483431609bbdcc33e2ed4149b85d56bcf169d4d_1_690x388.jpg)

>This one, right before she casts the spell, we can note some broken textures in the darker area, interestingly, they’re not even rendered correctly in Mednafen Saturn.

![img7](https://forums.libretro.com/uploads/default/optimized/2X/3/34cdc7cb2e76ee2a47830e1435d811bd5ec40521_1_690x388.jpg)

>Again, right spell, wrong aura

![img8](https://forums.libretro.com/uploads/default/optimized/2X/8/87fee3e90a11f2068fcd8a98e5224f7482ee44a7_1_690x388.jpg)

![img9](https://forums.libretro.com/uploads/default/optimized/2X/c/c37231685d9dd5bdebee6d81fff6920f9a0e6248_1_690x388.jpg)

>Both character cursors in battle or menus are rendered correctly

![img10](https://forums.libretro.com/uploads/default/optimized/2X/1/1161cd3d097e666a33657ed1738deedd41bc2154_1_690x388.jpg)

**Comment by LuismaSP89** [comment]
@barbudreadmon Awesome report ; )

**Comment by barbudreadmon** [comment]
@FCare here is the savestate you requested (Masqurin's turn and she is at spell range) : 
[Shining Force III (USA).zip](https://github.com/FCare/Kronos/files/2703584/Shining.Force.III.USA.zip)

Also, the right spell animation is available at 5:45 there : https://www.youtube.com/watch?v=25Lu-L_cMuQ

**Comment by FCare** [comment]
At least some of the graphic glitches were due to the framebuffer direct write access.
This access is fixed, it could be nice to test and update again the ticket.

**Comment by FCare** [comment]
No news since 6 days, I am closing the ticket. Reopen if there is still issues.

**Comment by barbudreadmon** [comment]
Currently playing SF3, and the issue with Masqurin fire spell reappeared, here is the savestate (Masqurin's turn and she is at spell range) : 
[Shining Force III (USA).zip](https://github.com/FCare/Kronos/files/2731894/Shining.Force.III.USA.zip)

In this battle i also think the floor textures are somehow broken.

**Comment by UncleRibs** [comment]
Same here, we are playing Shining Force 3 with the last version of Kronos and there are some FX glitches (mostly on fire based spells, Blaze, Phoenix invocation...).

If we can gather more detailed information for help, just ask for it.

**Comment by FCare** [comment]
All savestates are welcomed with screenshot of wht is expected vs what is
seen.
An important point also is to know your graphical card and the OS used.

**Comment by UncleRibs** [comment]
Hi,

Here are the information you have been asking for:

- OS : https://cdn.discordapp.com/attachments/469882022828048425/543034560062160902/unknown.png
- Graphical card : https://cdn.discordapp.com/attachments/469882022828048425/543034626336096256/unknown.png
- Save state is attached (Masculine about to cast Blaze) but graphical glitch happens on most of the heavy FX based spells (we have not been able to see any invocation correctly at the moment)
- Screenshot of what we see when the Blaze lvl 1 spell is cast : https://cdn.discordapp.com/attachments/469882022828048425/543035683309355008/unknown.png
- Screenshot of what we should see when the Blaze lvl 1 spell is cast : https://cdn.discordapp.com/attachments/469882022828048425/543366060401885184/unknown.png

Finally, you'll find the save state attached.

[GS-9188_000.zip](https://github.com/FCare/Kronos/files/2844480/GS-9188_000.zip)

-> Don't hesitate if you have any question or you need any more datas.

Cheers,

**Comment by UncleRibs** [comment]
We just tested to launch the game with the Windows 7 compatibility mode but the same graphical issues happen.

**Comment by barbudreadmon** [comment]
The part of the fire spell issue where things disappear and rectangles appear seems to be fixed in the WIP renderer branch. 
At the moment it seems only the fire animation shown at the left of the following screenshot around the foes is missing : 
![Masqurin](https://cdn.discordapp.com/attachments/469882022828048425/543366060401885184/unknown.png)

It appears there is currently a new issue where houses's shadows are broken though :
![shining force iii - scenario 1 - the titan of aspia usa t-en by shining force central v13 n -190303-092808](https://user-images.githubusercontent.com/10391265/53692919-3d0b8800-3d98-11e9-8207-aea5298d41d4.png)
They should be darker, not lighter.

@FCare Btw, could you reopen the issue ?

**Comment by FCare** [comment]
Sure I will reopen. Can you add savestates please?

Le dim. 3 mars 2019 à 09:42, barbudreadmon <notifications@github.com> a
écrit :

**Comment by barbudreadmon** [comment]
I don't have a working savestate for standalone (for whatever reason the libretro savestates aren't compatible with standalone, i need to figure out what's wrong with them), however i can post a 4M memory cart backup with a save in this town : 
[bkcartram.zip](https://github.com/FCare/Kronos/files/2922996/bkcartram.zip)

You can also see the fire spell issue in the next battle (go through the broken barrier, then the grotto), i think the previous savestates in this issue should also still be valid to show this issue.

**Comment by barbudreadmon** [comment]
Some update about this issue after the latest commits on the "rework" branch : 
- The blue layer introduced by a recent commit is gone !
- No "light" shadows anymore, actually they disappeared totally, they should look like the following screenshot though
![Shining Force III - Scenario 1 - The Titan of Aspia (USA)  T-En by Shining Force Central v13   n -190317-104518](https://user-images.githubusercontent.com/10391265/54488426-fc386680-48a1-11e9-9c90-6e7add186cd3.png)
- The fire animation (see previous screenshot) is still missing from masqurin's spells

I consider the 2 remaining issues as minor, missing graphics are way better than wrong graphics.
Previously posted backup ram and/or savestates should still be valid to test those bugs.

**Comment by barbudreadmon** [comment]
House shadows fixed with latest commits from the rework branch, only one issue is remaining : the missing spell effect on targets, and unlike my previous statement (i only tested masqurin because there was previously some issue with her spells), that's not only masqurin's fire spells, they are all missing.

None of the previous savestates seemed to work for me, some were totally denied by the emulator, some other caused inexpected crashes or freeze after loading (one of them actually triggered a saturn error message about general error instruction, asking me to cleanup my optical media), i'll post a new one tomorrow.

**Comment by barbudreadmon** [comment]
Done with "Shining Force III (USA)" : [MK-81383_001.zip](https://github.com/FCare/Kronos/files/3035814/MK-81383_001.zip)

**Comment by barbudreadmon** [comment]
New issues with SF3 : 
- the blue background in the panels disappeared (sometimes it's partly there)
- the house shadows disappeared again

**Comment by FCare** [comment]
I fixed the shadows again :-)

Le mar. 23 avr. 2019 à 18:02, barbudreadmon <notifications@github.com> a
écrit :

**Comment by barbudreadmon** [comment]
Thanks !

I noticed something interesting : reverting https://github.com/FCare/Kronos/commit/6a92d818edb0acef84d617607d1512d216fa73bd again fix the missing blue background panel issue, maybe the first time you removed the stencil clear operations was too early but now it's somehow needed ?

**Comment by barbudreadmon** [comment]
Latest commit didn't fix the missing spell effect issue. Also, the previously posted savestates aren't working anymore, and i can't create a new one : screen freeze after loading the savestate, actually it even crashed my desktop environment when i tried loading it early (bios).

Shortest way to reproduce the missing spell effect issue without savestate would be : 
- to use https://github.com/FCare/Kronos/files/2922996/bkcartram.zip as 4M memcard backup
- go out of the town through the **unbroken** barrier
- there is a grotto near some ruin (it isn't that far), go in there
- skip dialogs and characters turn until you can play a caster, most likely it will be irene and you'll be able to cast her heal on a neighbor : during the "battle" animation the "pillar of lights" that should be around her target are missing, they should be similar-looking to the following screenshot : 
![SF3 heal](http://www.rpgfan.com/reviews/shiningforce3/shinforce3-2.jpg)

PS : also, the panel's blue backgroung were fixed at 1x, but are still buggy at higher res

**Comment by CloudFR** [comment]
Hi, i am new user for github.
I have experience of emulation with kronos 1.6.0 for SF3 game scénario 1.
I have too graphical bugs (spell, summon, heal) in the battle.
Where are the fix ?
It's very pity because this emulation with kronos and SF3 with enhancement video like HD it's just amazing !
Good job !
